### PR TITLE
Closure killer iterate segment lanes

### PR DESCRIPTION
**Pull request by DaEgi01**
Killed IterateSegmentLanes with struct enumerators to simplify code and remove heap allocations.

### CODE REVIEW

**Review by krzychu124 — APPROVED** [review]
Looks good

**Comment by kianzarrin** [comment]
is this code used in performance critical part of the code?

**Comment by DaEgi01** [comment]
> is this code used in performance critical part of the code?

I think so.

**Review by kvakvs — APPROVED** [review]
Nice

File: TLM/Benchmarks/GetSegmentLaneIdAndSegmentPerfTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                m_nextLane = 2
+            },
+            new NetLane() {
+                m_nextLane = 0
+            },
+        };
+
+        [Benchmark]
+        public void Foreach_Direct() {
+            foreach (var item in new GetSegmentLaneIdsEnumerable(0, testLanes.Length, testLanes)) {
```
[the remark this answers is not in the document]
Comment (DaEgi01) [reply TLM/Benchmarks/GetSegmentLaneIdAndSegmentPerfTests.cs:22]: what do you mean?

[the remark this answers is not in the document]
Comment (kvakvs) [reply TLM/Benchmarks/GetSegmentLaneIdAndSegmentPerfTests.cs:22]: Empty loop body could be optimized out? Not sure how C# compiler handles empty loops

[the remark this answers is not in the document]
Comment (krzychu124) [reply TLM/Benchmarks/GetSegmentLaneIdAndSegmentPerfTests.cs:22]: Foreach won't be optimized because it may have side effects

**Pull request merged**